src/events.ts

```diff
@@ -150,11 +152,17 @@ const showQrCode = async (
   const alertConfig: AlertConfig = {
     title: 'Pair with Wallet',
     confirmButtonText: 'Done',
+    actionButtonText: 'Connect Wallet',
     body: `${getQrData(
       dataString,
       'svg'
-    )}<p>Don't know what to do with this QR code? <a href="https://docs.walletbeacon.io/supported-wallets.html" target="_blank">Learn more</a>.</p>`,
-    confirmCallback: () => undefined
+    )}<p>Connect wallet by scanning the QR code or clicking the link button <a href="https://docs.walletbeacon.io/supported-wallets.html" target="_blank">Learn&nbsp;more</a></p>`,
+    confirmCallback: () => undefined,
+    actionCallback: async () => {
+      const base58encoded = await serializer.serialize(data)
+      const uri = `web+tezos://?type=tzip10&data=${base58encoded}`
+      window.open(uri, '_blank')
```

Comment:
Theoretically, any web app can register a custom handler, then it's probably better to:
```
var otherWindow = window.open();
otherWindow.opener = null;
otherWindow.location = uri;
```
https://medium.com/@jitbit/target-blank-the-most-underestimated-vulnerability-ever-96e328301f4c

Reply:
That web apps need to ask for permission to become protocol handler should reduce most of the risk. But I agree that it's best to remove the back reference as you suggested.